tests/datasource/fluent/test_snowflake_datasource.py

```diff
@@ -0,0 +1,53 @@
+from __future__ import annotations
+
+import pydantic
+import pytest
+
+from great_expectations.datasource.fluent.config_str import ConfigStr
+from great_expectations.datasource.fluent.snowflake_datasource import (
+    SnowflakeDatasource,
+    SnowflakeDsn,
+)
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "connection_string, connect_args",
+    [
+        pytest.param(
+            "snowflake://<user_login_name>:<password>@<account_identifier>",
+            {"account": "my_account", "user": "my_user", "password": "123456"},
+            id="both connection_string and connect_args",
+        ),
+        pytest.param(None, {}, id="neither connection_string nor connect_args"),
+        pytest.param(
+            None,
+            {"account": "my_account", "user": "my_user"},
+            id="incomplete connect_args",
+        ),
+    ],
+)
+def test_conflicting_connection_string_and_args_raises_error(
+    connection_string: ConfigStr | SnowflakeDsn | None, connect_args: dict
+):
+    with pytest.raises(ValueError):
+        _ = SnowflakeDatasource(connection_string=connection_string, **connect_args)
+
+
+@pytest.mark.unit
+@pytest.mark.parametrize(
+    "connection_string",
+    [
+        pytest.param(
+            "not_snowflake://<user_login_name>:<password>@<account_identifier>",
+            id="bad scheme",
+        ),
+        pytest.param(
+            "snowflake://<user_login_name>@<account_identifier>", id="bad password"
+        ),
+        pytest.param("snowflake://<user_login_name>:<password>", id="bad domain"),
+    ],
+)
+def test_invalid_connection_string_raises_dsn_error(connection_string: str):
+    with pytest.raises(pydantic.ValidationError):
+        _ = SnowflakeDatasource(connection_string=connection_string)
```

Comment:
You are getting `name` errors here. I would introspect the pydantic error to ensure it's throwing the intended error.
Will update with an example.